### PR TITLE
File downloading fixes

### LINKED ISSUE #1058: Крашится сервер при попытке передать файлы
Когда клиент подключается к серверу где не настроен FastDL, то сервер крашится с этой ошибкой
```
Mem_Alloc: out of memory (alloc at D:\a\xash3d-fwgs\xash3d-fwgs\engine\common\net_chan.c:542)
```

### PR DESCRIPTION
**Pull request by SNMetamorph**
Fixes #1058 and #829

### CODE REVIEW

File: engine/common/net_chan.c [hunk 1]
```diff
@@ -978,8 +978,12 @@ int Netchan_CreateFileFragments( netchan_t *chan, const char *filename )
 	if( compressedFileTime >= fileTime )
 	{
 		// if compressed file already created and newer than source
-		if( FS_FileSize( compressedfilename, false ) != -1 )
+		int compressedSize;
+		compressedSize = FS_FileSize( compressedfilename, false );
+		if( compressedSize != -1 ) {
```
Comment (a1batross) [inline engine/common/net_chan.c:985]:
```
int compressedSize = FS_FileSize( compressedfilename, false );
if( compressedSize != -1 )
{
```

**Review by a1batross — CHANGES_REQUESTED** [review]

File: engine/common/net_chan.c (hunk 1 above)
Comment (a1batross) [reply engine/common/net_chan.c:985]: Sorry, it's fs_offset_t than int

Comment (SNMetamorph) [reply engine/common/net_chan.c:984]: Ok, fixed

Comment (a1batross) [reply engine/common/net_chan.c:984]: @SNMetamorph it wasn't though?

Comment (SNMetamorph) [reply engine/common/net_chan.c:984]: Why?

Comment (a1batross) [reply engine/common/net_chan.c:984]: { is not on new line

Comment (SNMetamorph) [reply engine/common/net_chan.c:984]: ah

Comment (SNMetamorph) [reply engine/common/net_chan.c]: It's fixed now :)

Comment (SNMetamorph) [reply engine/common/net_chan.c]: It's fixed now :)

File: engine/client/cl_main.c [hunk 2]
```diff
@@ -2472,11 +2472,18 @@ void CL_ProcessFile( qboolean successfully_received, const char *filename )
 	{
 		if( filename[0] != '!' )
 			Con_Printf( "processing %s\n", filename );
+
+		if( Q_strlen( filename ) > 11 && Q_stristr( filename, "downloaded/" ) == filename )
```
Comment (a1batross) [inline engine/client/cl_main.c:2476]: We really should at some point get rid of /downloaded folder and use `gamedir_download` like GoldSrc does...

Also, this is wrong, use `Q_strnicmp` if you want to compare the beginning of string.

**Review by a1batross — CHANGES_REQUESTED** [review]

File: engine/client/cl_main.c (hunk 2 above)
Comment (SNMetamorph) [reply engine/client/cl_main.c]: Ok replaced with `Q_strncmp`

**Pull request merged**